himbaechel/uarch/gowin/globals.cc

```diff
@@ -40,6 +40,8 @@ struct GowinGlobalRouter
 
     GowinGlobalRouter(Context *ctx) : ctx(ctx) { gwu.init(ctx); };
 
+    bool checkPipAvail(PipId pip) const { return gwu.is_global_pip(pip) || ctx->checkPipAvail(pip); };
```

Comment:
Forcing a PIP to be always available is likely to cause nasty problems, are you sure you want to be doing this?

Reply:
Ah, my error, this is inside the global router not the himbaechel API implementation, should be okay, but can you give this function a different name so it doesn't look like the nextpnr arch one? (say `global_pip_available`)

Reply:
You're right, now I myself don't like their similarity. I'll change it.

Reply:
Plus I broke the style rules, it was sloppy. :(